src/System.ServiceModel.Syndication/tests/BasicScenarioTests.cs

```diff
@@ -368,43 +358,130 @@ public static void AtomEntryPositiveTest_write()
         }
 
         [Fact]
-        [ActiveIssue(25058)]
-        public static void AtomFeedPositiveTest_WsFeedIcon()
+        public static void AtomFeedPositiveTest()
         {
-            string file = "ws-feed-icon.xml";
-            AtomFeedPositiveTest(file);
+            string dataFile = @"atom_feeds.dat";
+            List<string> fileList = GetTestFilesForFeedTest(dataFile);
+            List<AllowableDifference> allowableDifferences = GetAtomFeedPositiveTestAllowableDifferences();
+
+            foreach (string file in fileList)
+            {
+                FeedPositiveTest(file, FeedFormatterType.Atom10FeedFormatter, allowableDifferences);
```

Comment:
Instead of passing the formatter type, we can pass a Func, could be named `feedFormatter`, to the method. For example,

```c#
FeedPositiveTest(file, (o) =>new Atom10ItemFormatter(o), allowableDifferences);
```

And then we can use the Func in `FeedPositiveTest`. This avoids checking the enum types or argument checks in `FeedPositiveTest`.

Reply:
I have modified the code as you suggested.